### PR TITLE
[Bug] Fix web dependency for single file downloads

### LINKED ISSUE #40: web dependency doesn't work on single files
when downloading single files (like a binary from github) it fails to create the ext_dep folder

### PR DESCRIPTION
**Pull request by matthewfcarlson**
Fixes #40

### CODE REVIEW

**Comment by matthewfcarlson** [comment]
Thanks to @out0xb2 for finding the fix for this

File: edk2toolext/tests/test_web_dependency.py [hunk 1]
```diff
@@ -97,7 +124,7 @@ def test_get_internal_path_root_with_subfolders(self):
         second_level_dir_name = "second_dir"
         inner_dir_path = os.path.join(outer_dir, first_level_dir_name)
         self.assertEqual(WebDependency.get_internal_path_root(outer_dir,
-                         os.path.join(first_level_dir_name, second_level_dir_name)), inner_dir_path)
+                                                              os.path.join(first_level_dir_name, second_level_dir_name)), inner_dir_path)
```
Comment (out0xb2) [inline edk2toolext/tests/test_web_dependency.py]: Is this whitespace change of value?

Comment (matthewfcarlson) [reply edk2toolext/tests/test_web_dependency.py]: Flake errors

Comment (matthewfcarlson) [reply edk2toolext/tests/test_web_dependency.py]: We can't have a line over 120 chars, which it used to be longer?

File: edk2toolext/tests/test_web_dependency.py [hunk 2]
```diff
@@ -81,6 +94,20 @@ def test_fail_with_bad_url(self):
...
+    def test_single_file(self):
+        ext_dep_file_path = os.path.join(test_dir, "good_ext_dep.json")
+        with open(ext_dep_file_path, "w+") as ext_dep_file:
+            ext_dep_file.write(single_file_json_file)
+
+        ext_dep_descriptor = EDF.ExternDepDescriptor(ext_dep_file_path).descriptor_contents
+        ext_dep = WebDependency(ext_dep_descriptor)
+        ext_dep.fetch()
+
+        file_path = os.path.join(test_dir, "test_extdep", "test.txt")
```
Comment (spbrogan) [inline edk2toolext/tests/test_web_dependency.py]: would it be better to get test_extdep and test.txt from the "single_file_json_file" string?

File: edk2toolext/tests/test_web_dependency.py [hunk 3]
```diff
@@ -32,6 +32,19 @@
   "sha256":"68f2335344c3f7689f8d69125d182404a3515b8daa53a9c330f115739889f998"
 }
 '''
+# JSON file that describes a single file to download from the internet
+# Google.com was choosen as it's probably not going anywhere soon and it's small.
```
[the remark this answers is not in the document]
Comment (matthewfcarlson) [reply edk2toolext/tests/test_web_dependency.py]: Fixed

File: edk2toolext/tests/test_web_dependency.py [hunk 4]
```diff
@@ -81,6 +94,20 @@ def test_fail_with_bad_url(self):
             ext_dep.fetch()
             self.fail("should have thrown an Exception")
 
+    # try to download a single file from github
```
[the remark this answers is not in the document]
Comment (matthewfcarlson) [reply edk2toolext/tests/test_web_dependency.py]: Yeah that's a good point

File: edk2toolext/tests/test_web_dependency.py (hunk 2 above)
Comment (matthewfcarlson) [reply edk2toolext/tests/test_web_dependency.py]: yeah but since it's a string, we don't have a good way of reading is as yaml. I'll fix it

Comment (matthewfcarlson) [reply edk2toolext/tests/test_web_dependency.py]: Fixed

**Pull request merged**